### PR TITLE
metasploit: update to 6.0.12, add updateScript

### PR DESCRIPTION
**Pull request by makefu**
###### Motivation for this change

As a contribution to this years hacktoberfest i want to bump metasploit to the latest version.
Additionally I converted the maintainer comments for updating into a working shell script and added it via `passthru.updateScript`. I hope `nixpkgs-update` run by @r-ryantm will now pick up new versions automatically.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
  - before: `/nix/store/v3fqbxmscrgr0dy456cn9adq8sf78c93-metasploit-framework-5.0.90	 1028103384`
  - after: `/nix/store/dixmayl4s2kddr0yql3zrgs1fjvl9gms-metasploit-framework-6.0.4	 1012865360`
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
- [x] Tested updateScript via `nix-shell maintainers/scripts/update.nix --argstr path metasploit`

### CODE REVIEW

**Comment by makefu** [comment]
@GrahamcOfBorg build metasploit

**Comment by teto** [comment]
when starting I get
results/metasploit/bin/msfconsole
Following files may not be writable, so sudo is needed:
  /nix/store/g57ln13r4y0z374im449lh56wzylzkqz-metasploit-bundler-env/lib/ruby/gems/2.6.0
  /nix/store/g57ln13r4y0z374im449lh56wzylzkqz-metasploit-bundler-env/lib/ruby/gems/2.6.0/bin
  /nix/store/g57ln13r4y0z374im449lh56wzylzkqz-metasploit-bundler-env/lib/ruby/gems/2.6.0/bin
  /nix/store/g57ln13r4y0z374im449lh56wzylzkqz-metasploit-bundler-env/lib/ruby/gems/2.6.0/build_info
  /nix/store/g57ln13r4y0z374im449lh56wzylzkqz-metasploit-bundler-env/lib/ruby/gems/2.6.0/build_info/nokogiri-1.10.10.info
  /nix/store/g57ln13r4y0z374im449lh56wzylzkqz-metasploit-bundler-env/lib/ruby/gems/2.6.0/build_info/pg-1.2.3.info
  /nix/store/g57ln13r4y0z374im449lh56wzylzkqz-metasploit-bundler-env/lib/ruby/gems/2.6.0/build_info/sqlite3-1.4.2.info
  /nix/store/g57ln13r4y0z374im449lh56wzylzkqz-metasploit-bundler-env/lib/ruby/gems/2.6.0/bundler
  /nix/store/g57ln13r4y0z374im449lh56wzylzkqz-metasploit-bundler-env/lib/ruby/gems/2.6.0/cache
  /nix/store/g57ln13r4y0z374im449lh56wzylzkqz-metasploit-bundler-env/lib/ruby/gems/2.6.0/doc
  /nix/store/g57ln13r4y0z374im449lh56wzylzkqz-metasploit-bundler-env/lib/ruby/gems/2.6.0/extensions
  /nix/store/g57ln13r4y0z374im449lh56wzylzkqz-metasploit-bundler-env/lib/ruby/gems/2.6.0/gems
  /nix/store/g57ln13r4y0z374im449lh56wzylzkqz-metasploit-bundler-env/lib/ruby/gems/2.6.0/specifications

**Comment by ryantm** [comment]
Sounds like https://github.com/NixOS/nixpkgs/issues/101479

File: pkgs/tools/security/metasploit/update.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+latest=$(curl https://github.com/rapid7/metasploit-framework/releases.atom | xmlstarlet sel -N atom="http://www.w3.org/2005/Atom" -t -m /atom:feed/atom:entry -v atom:title -n | head -n1)
+echo "Updating metasploit to $latest"
+
+sed -i "s#refs/tags/.*#refs/tags/$latest\"#" Gemfile
+
+bundler install
+bundix
+sed -i '/[ ]*dependencies =/d' gemset.nix
+
+cd "../../../../"
```
Comment (Mic92) [inline pkgs/tools/security/metasploit/update.sh:16]: Is our `$PWD` always fixed?

Comment (makefu) [reply pkgs/tools/security/metasploit/update.sh:16]: it is after `cd "$(dirname "$(readlink -f "$0")")"`

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101687` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>metasploit</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101687` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>metasploit</li>
  </ul>
</details>

**Comment by makefu** [comment]
thanks!

**Pull request merged**